srv/params.go

```diff
@@ -274,6 +274,18 @@ func (p ParamsGetPegnetBalances) ValidChainID() *factom.Bytes32 {
 	return nil
 }
 
+type ParamsGetGraded struct {
+	Height int32 `json:"height,omitempty"`
+}
+
+func (p ParamsGetGraded) HasIncludePending() bool { return false }
+func (p ParamsGetGraded) IsValid() error {
+	return nil
```

Comment:
Negative height could be validated

Reply:
To maintain consistency across the API, I am adding a negative height check for every API method where height is a parameter. In so doing, I noticed that there is an activation height check  in the `get-bank` handler. 

https://github.com/pegnet/pegnetd/blob/master/srv/methods.go#L97-L99

Is this something I should replicate and, if so, should I move them all out into the parameter validation methods? The check seems to prevent anyone using this endpoint on a testnet. I believe it ought to either be more intelligent about detecting whether it is in use on mainnet or we should get rid of it. What do you think?

Reply:
I think that is your call. The error there is just a detailed error as the bank information isn't guaranteed to be populated in the db until that activation height.

`pegnetd` is implemented without testnet in mind, if you want to run it on testnet these flags are provided: https://github.com/pegnet/pegnetd/blob/master/cmd/root.go#L41-L45

The flags are mainly for local nets, rather than a consistent testnet.

Reply:
I'm happy to leave the bank handler as it is and add an activation height check to the `get-grades` endpoint. As both of the approaches work, I'd rather prioritise achieving consistency between handlers over fine tuning behaviour.

Reply:
The bank was a feature that didn't exist until block X, whereas the "graded" set has existed since inception. I don't think it's necessary to check for an activation height in this one but if you do, it's node.PegnetActivation.

Reply:
On second thought, if i were to add activation height checks then i'd have to add them to every handler, which could break some client code if it's relying on the presence of an empty array in the response. I will add the negative height check then we can think about more fine-grained validation another time.